data-prepper-plugins/rds-source/src/test/java/org/opensearch/dataprepper/plugins/source/rds/utils/IdentifierShortenerTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rds.utils;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.mockito.Mockito.mockStatic;
+
+class IdentifierShortenerTest {
+
+    @Test
+    void shortenIdentifier_when_input_is_shorter_than_max_then_return_original_input() {
+        final int maxLength = 10;
+        final String testInput = UUID.randomUUID().toString().substring(0, maxLength);
+
+        final String result = IdentifierShortener.shortenIdentifier(testInput, maxLength);
+
+        assertThat(result, equalTo(testInput));
+    }
+
+    @Test
+    void shortenIdentifier_when_input_is_longer_than_max_then_return_shortened_result() {
+        final int maxLength = 5;
+        final String testInput = UUID.randomUUID().toString();
+        assertThat(testInput.length(), greaterThan(maxLength));
+
+        final String result = IdentifierShortener.shortenIdentifier(testInput, maxLength);
+
+        assertThat(result.length(), lessThanOrEqualTo(maxLength));
```

Comment:
should this be `assertThat(result.length(), equalTo(testInput));`

Reply:
`testInput` is longer than maxLength (5) here. This is to verify that after shortening, it's equal or less than maxLength.

Reply:
I had typo.. `equalTo(maxLength))`

Reply:
Verify if there are no collisions when shortening the identifier.

Reply:
Did some smoke testing with long and similar inputs shortened to 10 chars, no collisions. With 15 chars used in rds source, the chance of collision should be very slim.